design/0001-metallb-crd.md

```diff
@@ -0,0 +1,251 @@
+# MetalLB Custom Resource Controller
+
+<!-- toc -->
+
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+- [Proposal](#proposal)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Initial CRD definition](#initial-crd-definition)
+- [Testing](#testing)
+- [Debugging and Troubleshooting](#debugging-and-troubleshooting)
+
+<!-- /toc -->
+
+## Summary
+
+The purpose of this enhancement is to be able to use kubernetes Custom
+Resource Definition "CRD" to configure MetalLB layer2 and BGP.
```

Comment:
Can you clarify that the proposal is to support custom resources as an alternative configuration interface?  I don't think we want to drop the existing ConfigMap interface (at least for now).

Something like ...

The purpose of this enhancement is to propose an alternative configuration interface for MetalLB based on Kubernetes Custom Resource Definitions (CRDs) while also retaining support for the existing ConfigMap interface.

Reply:
sure